From: Jason Dillaman <dillaman@redhat.com>
Date: Tue, 17 May 2016 16:14:42 -0400
Subject: rbd-mirror: lock ordering issue in status update callback

Signed-off-by: Jason Dillaman <dillaman@redhat.com>
(cherry picked from commit 5c04c1361074ca9e9df1bc1c2dd82dfafc2a1304)
(cherry picked from commit 9c7977da45aef44160efadb48cb2d606568d4a7b)
---
 src/tools/rbd_mirror/ImageSync.cc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/tools/rbd_mirror/ImageSync.cc b/src/tools/rbd_mirror/ImageSync.cc
index 2c6f5aa..24202f4 100644
--- a/src/tools/rbd_mirror/ImageSync.cc
+++ b/src/tools/rbd_mirror/ImageSync.cc
@@ -210,6 +210,8 @@ void ImageSync<I>::handle_copy_image(int r) {
 
 template <typename I>
 void ImageSync<I>::send_copy_object_map() {
+  update_progress("COPY_OBJECT_MAP");
+
   m_local_image_ctx->snap_lock.get_read();
   if (!m_local_image_ctx->test_features(RBD_FEATURE_OBJECT_MAP,
                                         m_local_image_ctx->snap_lock)) {
@@ -218,8 +220,6 @@ void ImageSync<I>::send_copy_object_map() {
     return;
   }
 
-  update_progress("COPY_OBJECT_MAP");
-
   assert(m_local_image_ctx->object_map != nullptr);
 
   assert(!m_client_meta->sync_points.empty());
